### ISSUE TITLE
Subscription breaks during performance test

### ISSUE THREAD

**Issue by JinseonyLee** [moquette-io/moquette#505]
Hi, 

I tested mqtt server with moquette 0.12.1 version with H2.

I sent 1 message/per sec with 2 ~ 3000 clients.

When I first started testing, then It was done properly.
But, when all clients are disconnected and restarted testing, It can't use internalPublish methods.

It is the error log below.

```
Caused by: java.lang.NullPointerException: null
        at io.moquette.broker.subscriptions.CTrie.evaluate(CTrie.java:58)
        at io.moquette.broker.subscriptions.CTrie.recursiveMatch(CTrie.java:70)
        at io.moquette.broker.subscriptions.CTrie.recursiveMatch(CTrie.java:86)
        at io.moquette.broker.subscriptions.CTrie.recursiveMatch(CTrie.java:86)
        at io.moquette.broker.subscriptions.CTrie.recursiveMatch(CTrie.java:86)
        at io.moquette.broker.subscriptions.CTrie.recursiveMatch(CTrie.java:65)
        at io.moquette.broker.subscriptions.CTrieSubscriptionDirectory.matchWithoutQosSharpening(CTrieSubscriptionDirectory.java:67)
        at io.moquette.broker.subscriptions.CTrieSubscriptionDirectory.matchQosSharpening(CTrieSubscriptionDirectory.java:72)
        at io.moquette.broker.PostOfficeImpl.publish2Subscribers(PostOfficeImpl.java:226)
        at io.moquette.broker.PostOfficeImpl.internalPublish(PostOfficeImpl.java:304)
```

I checked the subscription tree trace log. 

```
 subscription tree root
    |-level1
    | |-level2
    | | |-client.1 ~~[{filter=level1/level2/client.1, qos=AT_MOST_ONCE, client='level2+client.1'}]
    | | |-''TNode
    | | |-''TNode
    | | |-''TNode
```

When the exception occured, the subscription suddenly changed from CNode to TNode.
I added log to check Node type and token value in CTrie.java

```
CTrie.java(62line)       cnode = io.moquette.broker.subscriptions.TNode@1f, token = FirstTest.7
CTrie.java(63line)       cnode.token = null
Caused by: java.lang.NullPointerException: null
        at io.moquette.broker.subscriptions.CTrie.evaluate(CTrie.java:58)
        at io.moquette.broker.subscriptions.CTrie.recursiveMatch(CTrie.java:70
```

```
java -version
openjdk version "1.8.0_212"
OpenJDK Runtime Environment (Zulu 8.38.0.13-CA-linux64) (build 1.8.0_212-b04)
OpenJDK 64-Bit Server VM (Zulu 8.38.0.13-CA-linux64) (build 25.212-b04, mixed mode)

cat /etc/os-release 
NAME="Red Hat Enterprise Linux Server"
VERSION="7.5 (Maipo)"

```
Is there any way to solve this situation?

**Comment by JinseonyLee** [comment]
I found a solution. 
In performance test, it is processed in Tnode format in CTrie.java's recursiveMatch methods.
NullPointerException is not thrown if the source is changed as follows:
It just changes the order of processing.
Before
```   private Set<Subscription> recursiveMatch(Topic topic, INode inode) {
        CNode cnode = inode.mainNode();
        NavigationAction action = evaluate(topic, cnode);
        if (action == NavigationAction.MATCH) {
            return cnode.subscriptions;
        }
        if (action == NavigationAction.STOP) {
            return Collections.emptySet();
        }
        if (cnode instanceof TNode) {
            return Collections.emptySet();
        }
```
TO-BE
```   private Set<Subscription> recursiveMatch(Topic topic, INode inode) {
        CNode cnode = inode.mainNode();
        if (cnode instanceof TNode) {
        	return Collections.emptySet();
        }
        NavigationAction action = evaluate(topic, cnode);
        if (action == NavigationAction.MATCH) {
            return cnode.subscriptions;
        }
        if (action == NavigationAction.STOP) {
            return Collections.emptySet();
        }
```

**Comment by andsel** [comment]
Thanks @JinseonyLee good suggestion, I fixed it in PR #564
